src/huggingface_hub/commit_api.py

```diff
@@ -0,0 +1,424 @@
+"""
+Type definitions and utilities for the `create_commit` API
+"""
+
+import base64
+import io
+import os
+import sys
+from concurrent import futures
+from concurrent.futures import ThreadPoolExecutor
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+from typing import BinaryIO, Dict, Iterable, List, Optional, Tuple, Union
+
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
```

Comment:
Not really important, but curious, which one do you prefer? What's here or this suggestion?

```suggestion
try:
    from typing import Literal
except ImportError:
    from typing_extensions import Literal
```

Reply:
I prefer the `sys.version_info` solution to the try/except. idk just feels cleaner to me imo

Reply:
No preference for me, though the `sys.version_info` solution is consistent with the rest of the codebase